src/main/java/qupath/ext/wsinfer/ui/WSInferController.java

```diff
@@ -165,32 +166,34 @@ private void configureModelChoices() {
         WSInferModelCollection models = WSInferUtils.getModelCollection();
         modelChoiceBox.getItems().setAll(models.getModels().values());
         modelChoiceBox.setConverter(new ModelStringConverter(models));
-//        modelChoiceBox.getSelectionModel().selectedItemProperty().addListener((observable, oldValue, newValue) -> {
-//        });
-        forceRefreshButton.disableProperty().bind(modelChoiceBox.getSelectionModel().selectedItemProperty().isNull());
+        modelChoiceBox.getSelectionModel().selectedItemProperty().addListener((v, o, n) ->
+            n.isModelAvailableProperty().set(n.isValid()));
+        downloadButton.disableProperty().bind(
```

Comment:
From my previous comment, I don't think this should be a binding.
The previous behavior should have ensured there was a check every time the selected model changed (I think...), so that it could also pick up at that point if a model was deleted manually from the file system

Reply:
By "this", do you mean binding the disable property? That's fine by me, as long as we check SHAs at runtime it should avoid any weird errors from malformed files

Reply:
Also, the download method now checks if we have a valid model anyways, and if we do it just does nothing (except pop up a "model available" note, so disabling the download button is nonsensical anyways.

Do you mind the overhead of SHA256ing the model file whenever we check `isValid()`?

Reply:
Not really, but probably worth checking the timing for the largest models we've got (I expect it's fast enough to be negligible, right?).

Disabling the property is good when we have a valid model, I think - but that can be done explicitly when the model is selected from the choice box and not through a binding.

Reply:
The longest time for the SHA itself is 700ms on my CPU, which is 10 years old. Unlikely to be much beyond noise when compared to inference. Didn't time disk time as that'll vary a lot based on the type of SSD/HD people are using

Sure, as long as we check at runtime then we handle on-disk deletion between downloading and running

Reply:
Hmmm, seems borderline - probably completely fine in practice. The only scenario I can see where it would be bad is when using a large model on a few tiles, which I guess should be rare (and waiting a second isn't too much to ask).

The alternative I was thinking of was to [store the last modified time](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/nio/file/Files.html#getLastModifiedTime(java.nio.file.Path,java.nio.file.LinkOption...)) and only compute whenever there's reason to believe the file is more recent than the last SHA computation.

Is it clear when there's an attempt to make some https request?
What happens for models that are downloaded and there is no internet connection available?

Reply:
Also changed the cache directory to `PathPrefs.getUserPath() + File.separator + "wsinfer"` which should hopefully be sensible across-platform. Seems a bit antisocial to put the model files somewhere new if they've already set a global QuPath directory.

Reply:
Yeah, makes sense. Conceivably trouble if they have reset the user directory, so ought really to check for null (and maybe return to the previous default?).

(Not likely to happen in practice, since it would take some hacking to get wsinfer loaded as an extension... but could happen to devs working from an IDE, i.e. future us)

Reply:
Yeah, the user/extension path stuff needs looked at again at some point - https://github.com/qupath/qupath/issues/1260

Reply:
Well unless you've any other comments/concerns I think this is good to go, will test out actions to publish a release JAR now